qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,379 @@
+
+        """
+        if len(frequencies) < 3:
+            raise ValueError("Spectroscopy requires at least three frequencies.")
+
+        if unit not in self.__units__:
+            raise ValueError(f"Unsupported unit: {unit}.")
+
+        self._frequencies = [freq * self.__units__[unit] for freq in frequencies]
+        self._absolute = absolute
+
+        super().__init__([qubit], circuit_options=("amp", "duration", "sigma", "width"))
+
+    def circuits(self, backend: Optional["Backend"] = None, **circuit_options):
+        """
+        Create the circuit for the spectroscopy experiment. The circuits are based on a
+        GaussianSquare pulse and a frequency_shift instruction encapsulated in a gate.
+
+        Args:
+            backend: A backend object.
+            circuit_options: Key word arguments to run the circuits. The circuit options are
+                - amp: The amplitude of the GaussianSquare pulse, defaults to 0.1.
+                - duration: The duration of the GaussianSquare pulse, defaults to 1024 samples.
+                - sigma: The standard deviation of the GaussianSquare pulse, defaults to one
+                    fifth of the duration.
+                - width: The width of the flat top in the GaussianSquare pulse, defaults to 0.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+
+        Raises:
+            QiskitError: If relative frequencies are used but no backend was given.
+        """
+        if not backend and not self._absolute:
+            raise QiskitError("Cannot run spectroscopy relative to qubit without a backend.")
+
+        amp = circuit_options.get("amp", 0.1)
+        duration = circuit_options.get("duration", 1024)
+        sigma = circuit_options.get("sigma", duration / 4)
+        width = circuit_options.get("width", 0)
+
+        drive = pulse.DriveChannel(self._physical_qubits[0])
+
+        # Create a template circuit
+        freq_param = Parameter("frequency")
+        with pulse.build(name="spectroscopy") as sched:
+            pulse.set_frequency(freq_param, drive)
+            pulse.play(pulse.GaussianSquare(duration, amp, sigma, width), drive)
+
+        gate = Gate(name="Spec", num_qubits=1, params=[freq_param])
+
+        circuit = QuantumCircuit(1)
+        circuit.append(gate, (0,))
+        circuit.add_calibration(gate, (self._physical_qubits[0],), sched, params=[freq_param])
+        circuit.measure_active()
+
+        if not self._absolute:
+            center_freq = backend.defaults().qubit_freq_est[self._physical_qubits[0]]
+
+        # Create the circuits to run
+        circs = []
+        for freq in self._frequencies:
+            if not self._absolute:
+                freq += center_freq
+
+            assigned_circ = circuit.assign_parameters({freq_param: freq}, inplace=False)
+            assigned_circ.metadata = {
+                "experiment_type": self._type,
+                "qubit": self._physical_qubits[0],
```

Comment:
Is this `qubit` or `qubits:List[int]`? Other experiment like 2q RB may define this as `qubits` and we may need this logic in base curve fitter.
```python
if "qubits" in metadata:
    qubits = list(metadata["qubits"])
else:
   qubits = [metadata["qubit"]]
```
I don't prefer common field have multiple name representation. This is not only for this PR.

Reply:
This aligns with the property in the base class `def physical_qubits(self) -> Tuple[int]:`. I refactored this to use the property here: 6173531. If there is a package wide alignment to be done can you open an issue?

Reply:
https://github.com/Qiskit/qiskit-experiments/pull/5/files#r603392572